examples/Multi_Armed_Bandit/bernoulli_multi_armed_bandit.py

```diff
@@ -0,0 +1,175 @@
+# You can think of each arm as representing a version of the marketing campaign, and the corresponding win rates would be the true effect size of each marketing material version.
+
+# We can build such a black box system with just a few lines of code:
+
+
+@dataclass
+class MultiArmedBandit:
+    """A stateless black box system offering a finite number of actions (= arms)."""
+
+    win_rates: list[float]
+    """Win rates associated with the arms."""
+
+    def draw_arm(self, arm_index: int) -> bool:
+        """Draw a reward by pulling the specified arm."""
+        return bernoulli(self.win_rates[arm_index]).rvs()
+
+
+### Setup
+
+# For our example, we use a system with a fixed collection of win rates, which we
+# assume are unknown to us:
+
+WIN_RATES = [0.3, 0.4, 0.6, 0.65, 0.8]
+BANDIT = MultiArmedBandit(WIN_RATES)
+
+# Given this target system, we consider the following two tasks:
+# - **Online optimization:** Maximizing the total reward accumulated during a campaign
+# - **Active learning:** Estimating the win rates of all arms
+
+# Solving these task can be achieved through an appropriate choice of acquisition function:
+# For online optimization, we use {class}`~baybe.acquisition.acqfs.qThompsonSampling`.
+# For active learning, we switch to {class}`~baybe.acquisition.acqfs.PosteriorStandardDeviation`.
+
+# To estimate the corresponding effects, we simulate each campaign for a certain number of steps and repeat this process in multiple Monte Carlo runs:
+
+ACQFS = [
+    qThompsonSampling(),  # Online optimization
+    PosteriorStandardDeviation(),  # Active learning
+]
+N_MC_RUNS = 10
+N_ITERATIONS = 200
+
+
+### Building the Model
+
+# To model the rewards obtained from the bandit, we use a {class}`~baybe.targets.binary.BinaryTarget`.
+# The corresponding search space is spanned by a {class}`~baybe.parameters.categorical.CategoricalParameter` describing the bandit's arms.
+
+n_arms = len(WIN_RATES)
+objective = BinaryTarget(name="clicked").to_objective()
+searchspace = CategoricalParameter(
+    name="Bandit Arm", values=[f"arm_{i}" for i in range(n_arms)]
+).to_searchspace()
+surrogate = BernoulliMultiArmedBanditSurrogate()
+
+
+# For each simulation, we report the trajectory of earned rewards and the estimated win rates of the bandit arms:
+
+
+class SimulationResult(NamedTuple):
+    """Result of a bandit campaign simulation."""
+
+    earned_rewards: np.ndarray
+    estimated_win_rates: np.ndarray
+
+
+# Putting all pieces together, we define a function to simulate our campaign under a given acquisition strategy:
+
+
+def simulate(acqf: AcquisitionFunction) -> SimulationResult:
+    """Simulate the campaign with with the given acquisition function."""
+    recommender = TwoPhaseMetaRecommender(
+        initial_recommender=RandomRecommender(
+            allow_repeated_recommendations=True,
```

Comment:
Why setting all of these default flags here to true? Is this also necessary? If so, that should be mentioned here (and is this checked somewhere in the code?)

Reply:
Well, this is sort of a dilemma and demonstrates why I don't like our current metadata handling approach: because we alter the metadata in the searchspace object, the object is effectively mutable and cannot be reused, because in the second use, you'll encounter the already modified metadata. 

Setting the allow_flags here was sort of my workaround for that. The other way is to move the search space generation into the loop / create some `make_campaign` function. But in that case, I'm not able to nicely write docstrings that are rendered in our doc 😢 

So what do you suggest as solution?

Reply:
Well, we could just add information on that in the docstring of the `surrogates/bandits.py` file. Until we have a user guide for MAB, this would be the place where I'd collect this information. Most important point for me is just that user need to be made aware if setting these flags is currently necessary or not, and to have that information at some reasonable place. And if we expect another refactoring here, then I am also open to a temporary solution for this.

Reply:
I think you did not get my point, it has nothing to do with the MAB but is a problem with our search space. Nevermind:
* I changed the structure to solve the problem in another way
* Will open an issue on Github to remind us of the problem